mobile/library/java/org/chromium/net/CronvoyProvider.java

```diff
@@ -106,41 +100,21 @@ public String toString() {
         + "enabled=" + isEnabled() + "]";
   }
 
-  /** Name of the Java {@link CronetProvider} class. */
-  private static final String JAVA_CRONET_PROVIDER_CLASS =
-      "org.chromium.net.impl.JavaCronetProvider";
-
-  /** Name of the native {@link CronetProvider} class. */
-  private static final String NATIVE_CRONET_PROVIDER_CLASS =
-      "org.chromium.net.impl.NativeCronetProvider";
-
-  /** {@link CronetProvider} class that is packaged with Google Play Services. */
-  private static final String PLAY_SERVICES_CRONET_PROVIDER_CLASS =
-      "com.google.android.gms.net.PlayServicesCronetProvider";
+  /** Name of the native {@link CronvoyProvider} class. */
+  private static final String NATIVE_CRONVOY_PROVIDER_CLASS =
+      "org.chromium.net.impl.NativeCronvoyProvider";
 
   /**
-   * {@link CronetProvider} a deprecated class that may be packaged with some old versions of Google
-   * Play Services.
-   */
-  private static final String GMS_CORE_CRONET_PROVIDER_CLASS =
-      "com.google.android.gms.net.GmsCoreCronetProvider";
-
-  /**
-   * Returns an unmodifiable list of all available {@link CronetProvider}s. The providers are
+   * Returns an unmodifiable list of all available {@link CronvoyProvider}s. The providers are
    * returned in no particular order. Some of the returned providers may be in a disabled state and
-   * should be enabled by the invoker. See {@link CronetProvider#isEnabled()}.
+   * should be enabled by the invoker. See {@link CronvoyProvider#isEnabled()}.
    *
    * @return the list of available providers.
    */
-  public static List<CronetProvider> getAllProviders(Context context) {
+  public static List<CronvoyProvider> getAllProviders(Context context) {
     // Use LinkedHashSet to preserve the order and eliminate duplicate providers.
-    Set<CronetProvider> providers = new LinkedHashSet<>();
-    addCronetProviderFromResourceFile(context, providers);
-    addCronetProviderImplByClassName(context, PLAY_SERVICES_CRONET_PROVIDER_CLASS, providers,
-                                     false);
-    addCronetProviderImplByClassName(context, GMS_CORE_CRONET_PROVIDER_CLASS, providers, false);
-    addCronetProviderImplByClassName(context, NATIVE_CRONET_PROVIDER_CLASS, providers, false);
-    addCronetProviderImplByClassName(context, JAVA_CRONET_PROVIDER_CLASS, providers, false);
```

Comment:
Can you say more about this change? Is this what you mean about "user needs to ask CronvoyProvider instead of CronetProvider to provide a Cronet engine"?

Reply:
The removal of unnecessary providers is to make sure we are using EM under Cronvoy.
The change from CronetProvider to CronvoyProvider is because CronetProvider is where an application goes to get a CronetEngine.
By sharing the same CronetProvider name, the app doesn't know which provider, Cronet or Cronvoy, to call on when both libraries are present.
This change slightly modifies Cronvoy's API to make sure both Cronet and Cronvoy can live in the same app.